plaidml/bridge/openvino/ops/topk.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using Direction = edsl::SortDirection;
+using Mode = ngraph::opset4::TopK::Mode;
+using Sort = ngraph::opset4::TopK::SortType;
+
+namespace PlaidMLPlugin {
+
+void registerTopK() {
+  registerOp("topk", [](const Context& ctx) {
+    auto* layer = ngraph::as_type<ngraph::opset4::TopK>(ctx.layer);
+    IE_ASSERT(ctx.operands.size() == 2);
+    auto I = ctx.operands.at(0);
+    auto K = layer->get_k();
+    auto axis = layer->get_axis();
+    auto mode = layer->get_mode();
+    auto sort = layer->get_sort_type();
+    auto index_element_type = layer->get_index_element_type();
+    long ndims = I.rank();
+
+    edsl::Tensor values, indices;
+    Direction direction;
+    switch (mode) {
+      case Mode::MAX:
+        direction = Direction::DESC;
+        break;
+      case Mode::MIN:
+        direction = Direction::ASC;
+        break;
+      default:
+        THROW_IE_EXCEPTION << "invalid topk sort mode";
+    }
+    values = op::sort(I, axis, direction);
+    indices = edsl::argsort(I, axis, direction);
+
+    std::vector<edsl::Tensor> Os;
+    edsl::Tensor out_vals = edsl::gather(values, edsl::index({edsl::TensorDim(K)}, 0)).axis(axis);
```

Comment:
This seems to be only used in `case Sort::SORT_VALUES` probably it is better to move under that case.

Reply:
Sure.

Reply:
Sorry, I missed `Os.push_back(out_vals);` at line 75. I think `out_vals` should be seems globally.

Reply:
Yes, I mean this gather is only needed in `Sort::SORT_VALUES`

Reply:
OK. I rename those variables and move it under that case.